src/multicore.rs

```diff
@@ -139,16 +182,19 @@ mod implementation {
         }
     }
 
-    pub struct WorkerFuture<T, E> {
-        future: future::FutureResult<T, E>,
+    pub struct Waiter<T> {
+        val: Option<T>,
     }
 
-    impl<T: Send + 'static, E: Send + 'static> Future for WorkerFuture<T, E> {
-        type Item = T;
-        type Error = E;
+    impl<T> Waiter<T> {
+        /// Wait for the result.
```

Comment:
As far as I can see, this is inaccurate: `std::Option::take` will never wait.

Reply:
It won't, but this is emulating the threaded API. Both instances are effectively blocking, one just blocks for longer.